### PR TITLE
add gatk4/printreads

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

See title

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4/printreads/meta.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: "gatk4_printreads"
+description: Print reads in the SAM/BAM/CRAM file
+keywords:
+  - printreads
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/printreads/meta.yml:4]:
```suggestion
  - gatk4
  - bam
  - cram
  - sam
  - printreads
```

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/nf-core/gatk4/printreads/main.nf [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    } else {
+        avail_mem = task.memory.giga
+    }
+    """
+    gatk --java-options "-Xmx${avail_mem}g" PrintReads \\
+        $args \\
+        --reference $fasta \\
+        --input $input \\
+        --read-index $index \\
+        --output ${prefix}.reads.${input.getExtension()}
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/printreads/main.nf:40]: It's preferred to not put any custom extensions on the output files. The pipeline creator should add this themselves using `ext.prefix`. You could also add a check to see if the output file has the same name as the input file and if so, return an error (see [here](https://github.com/nf-core/modules/blob/3dd73937b084b547f9272bc901e0f120a7913fd8/modules/nf-core/bedtools/closest/main.nf#L31-L33))

File: modules/nf-core/gatk4/printreads/meta.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - bam:
+      type: file
+      description: BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
+  - bai:
+      type: file
+      description: BAM/CRAM/SAM file
```
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/gatk4/printreads/meta.yml:32]: Don't forget this one ;)

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Some small changes and then it's good to go!

File: modules/nf-core/gatk4/printreads/meta.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      description: Index of the sorted BAM file
+      pattern: "*.{bam}"
+  - crai:
+      type: file
+      description: Index of the Sorted CRAM file
+      pattern: "*.{cram}"
+  - sai:
+      type: file
+      description: Index of the sorted SAM file
+      pattern: "*.{sam}"
```
Comment (nvnieuwk) [inline modules/nf-core/gatk4/printreads/meta.yml:73]: You're outputting alignment files and not indices ;)

**Review by nvnieuwk — APPROVED** [review]
One last comment :rocket:

File: modules/nf-core/gatk4/printreads/main.nf (hunk 2 above)
Comment (ramprasadn) [reply modules/nf-core/gatk4/printreads/main.nf]: That's pretty neat! TIL! 

And btw, does bedtools/closest module throw an error for you when the input and output filenames are the same? I thought we had to use string interpolation (like `"${input_1}"` and `"${input_2}"`) to get the values of the input variables in the script block.. 🤔

File: modules/nf-core/gatk4/printreads/meta.yml (hunk 4 above)
Comment (ramprasadn) [reply modules/nf-core/gatk4/printreads/meta.yml]: Oops fixed 😅

File: modules/nf-core/gatk4/printreads/main.nf (hunk 2 above)
Comment (nvnieuwk) [reply modules/nf-core/gatk4/printreads/main.nf]: I haven't actually tested this tbh :grin: I copied that piece of code from another module. But you're probably right about it

File: modules/nf-core/gatk4/printreads/meta.yml (hunk 4 above)
Comment (nvnieuwk) [reply modules/nf-core/gatk4/printreads/meta.yml]: Awesome! Feel free to merge ;)

Comment (nvnieuwk) [reply modules/nf-core/gatk4/printreads/meta.yml]: oh nvm, you should use `bam`, `cram` and `sam` as output and not their indices

Comment (nvnieuwk) [reply modules/nf-core/gatk4/printreads/meta.yml]: at least that's what your module outputs

Comment (ramprasadn) [reply modules/nf-core/gatk4/printreads/meta.yml]: 👍🏻

**Comment by nvnieuwk** [comment]
Looks good now! You can hit the merge :rocket:

**Pull request merged**